docs/articles/user_acquisition_analytics.md

```diff
@@ -0,0 +1,448 @@
+
+So far, we know that:
+
+- many more users signed up in response to the influencer-backed ad creatives (second campaign) than the first campaign
+- which specific ad creatives generated the most user signups
+- the vast majority of users have low to moderate activity
+
+We don't know:
+
+- which ad creatives resulted in low vs medium vs high activity users?
+- how did users cluster in terms of signup date, activity levels, *and* ad_creatives they clicked?
+- are there user data patterns we can mine to make better decisions when planning our next campaign?
+
+Fortunately, embedding with Superlinked spaces will help provide answers to these questions, and empower us to adopt a more effective user acquisition strategy.
+
+## Embedding with Superlinked
+
+Now, let's use Superlinked to embed our data in a semantic space - to:
+ 
+1. inform our model as to which ad creatives generated signups, and *which users* (with specific activity level distributions) signed up
+2. group specific ad creatives that have similar meanings, and possibly outcomes
+
+First, we define a schema for our user data:
+
+```python
+class UserSchema(Schema):
+    ad_creative: String
+    activity: Float
+    signup_date: Timestamp
+    id: IdField
+```
+
+```python
+user = UserSchema()
+```
+
+Now we create a **semantic space** for our **ad_creatives** using a text similarity model. Then, we encode user activity into a **numerical space** to represent **users' activity level**. We also encode the **signup date** into a **recency space**, allowing our clustering algorithm to take account of the two specific periods of signup activity (following our two campaign start dates).
+
+```python
+# create a semantic space for our ad_creatives using a text similarity model
+creative_space = TextSimilaritySpace(
+    text=user.ad_creative, model="sentence-transformers/all-mpnet-base-v2"
+)
+
+# encode user activity into a numerical space to represent users' activity level
+activity_space = NumberSpace(
+    number=user.activity, mode=Mode.SIMILAR, min_value=0.0, max_value=1.0
+)
+
+# encode the signup date into a recency space
+recency_space = RecencySpace(
+    timestamp=user.signup_date,
+    period_time_list=[PeriodTime(timedelta(days=65)), PeriodTime(timedelta(days=185))],
+    negative_filter=0.0,
+)
+```
+
+Let's plot our recency scores by date. 
+
+```python
+recency_plotter = RecencyPlotter(recency_space, context_data=EXECUTOR_DATA)
+recency_plotter.plot_recency_curve()
+```
+
+![](../assets/use_cases/user_acquisition_analytics/recency_scores-by-date.png)
+
+Next, we set up an in-memory data processing pipeline for indexing, parsing, and executing operations on user data, including clustering (where RecencySpace lets our model take account of user signup recency).
```

Comment:
clustering happens outside the Superlinked framework. The executor does the vectorization according to our spaces grouped in the index, and the Vector Sampler will help us export them to work on them (clustering, umap)